agora-drep/Agora/Proxy.hs

```diff
@@ -254,9 +255,15 @@ proxyScript = plam $ \authSymbol' ctx -> P.do
                             _ -> pcon PFalse
                     )
 
+            let minted = psymbolValueOf # pfromData currencySymbol # pfromData mint
```

Comment:
I'm not 100% confident here cause if you can somehow burn 2 GATs with empty name (legitimate ones) in one tx and mint 1 new (with non-empty) you'd still get -1 total and get a phony GAT, but I'm not sure if that'd mean that the effects that were holding these 2 GATs were buggy or not.

Generally for performance anyway here we should extract the token map that corresponds to own symbol (we can't say anything about other symbols cause there may be effects that mint something) and pattern match that there is only entry with empty name and that the value is -1. You can mint only tokens with empty name in a legitimate way anyway so not having that fold over a map with presumably only one entry would be good